synapse/config/saml2_config.py

```diff
@@ -75,15 +73,81 @@ def read_config(self, config, **kwargs):
 
         self.saml2_enabled = True
 
-        self.saml2_mxid_source_attribute = saml2_config.get(
-            "mxid_source_attribute", "uid"
-        )
-
         self.saml2_grandfathered_mxid_source_attribute = saml2_config.get(
             "grandfathered_mxid_source_attribute", "uid"
         )
 
-        saml2_config_dict = self._default_saml_config_dict()
+        user_mapping_provider_dict = saml2_config.get("user_mapping_provider")
+        if user_mapping_provider_dict is None:
+            user_mapping_provider_dict = {}
+
+        # Load configured module class and config
+        default_mapping_provider = (
```

Comment:
given that this is a constant, (a) can it be at the top level rather than inside the class, (b) can it have a SHOUTY_NAME